### PR TITLE
Implemented devise

### PR DESCRIPTION
**Pull request by vangelif**
# In this PR, we: 

- Removed current_user method in ApplicationController.
- Installed devise.
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password is stored in the database.
- Asks for confirmation of email.
- Can reset password.
- Followed best practises, no linters errors

### CODE REVIEW

**Review by skyv26 — COMMENTED** [review]
## Hi @vangelif, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🟨 




## Reason 🗒️ 
You have submitted a project, that needs the contribution from your `pair-programming` partners, but it seems like only 2 people contributed to the project and 3rd person didn't contribute, so that's why I can't proceed with the review. You still have 1 working day including today, please write an email to Student Success. If student success allows you to submit your project by unpairing your 3rd partner or other possible scenario then as proof, you need to attach the email screenshot in the PR before asking for the review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Nemwel-Boniface — COMMENTED** [review]
# Hello @vangelif , @jubaer919  and @maths29 :wave: 

Thank you for making sure that the third partner has contributed commits to the project. However since the project was already done and completed, it is observed that the third partner has made "fake" commits since there was nothing left to be done, the project was ready for review.

**What you need to do as a team:**

- [ ] Please close this PR and create a new branch where you will work on this milestone afresh this time round ensuring that everyone has made a meaningful contribution to this milestone.

**Note to other reviewers to not review this PR.**

As always #happycoding  :rocket:

**Comment by enniopo** [comment]
For the next Reviewer: Please proceed with the review even if there are no commits from maths29. 
Student Success authorized @vangelif and @jubaer919 to submit the project. 

@vangelif - please submit another review request and the next CR will review your project.

**Comment by vangelif** [comment]
> For the next Reviewer: Please proceed with the review even if there are no commits from maths29. Student Success authorized @vangelif and @jubaer919 to submit the project.
> 
> @vangelif - please submit another review request and the next CR will review your project.

@enniopo - appreciate for the understanding here

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      <%= f.check_box :remember_me, class: "form-check-input", id: "exampleCheck1" %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in", class: "btn btn-warning my-3" %>
+  </div>
+<% end %>
+
```
Comment (thecodechaser) [inline app/views/devise/sessions/new.html.erb]: - [x] Great job on the project requirements but there's one style issue. Please make sure the login button has complete width and the "Log in" text is in the center to follow the wireframe design as requested in the project requirements.

### Your design

image

### Wireframe design

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @vangelif, and Team👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with authentication. ✔️ 
- Nice UI design. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @vangelif and @skyv26,

✔️ Well-centered form according to design

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**